CO-2750 Hr attendance management migration v11

quentingigon opened the pull request.

eicher31 committed:
Move hr_attendance_management from CompassionCH/hr

FIX flake8 error

fix for tests

fix plint

migration done, need testing now

module is installable

fix flake

corrections for tests and travis

added hr_holidays_public to oca_dependencies

fixed errors on check-in page

fix flake

fix for CRONs in v11 and fix for tests

further fixes for CRONs

test fix hr_holidays_public

fixed oca

fix flake and stuff

merged PR #1294 with this one

trying to fix tests

ecino commented:
hr_attendance_management/models/hr_employee.py:86:89: E501 line too long (92 > 88 characters)
@eicher31 please review

eicher31 commented on hr_attendance_management/data/annual_balance_cron.xml:
```diff
@@ -0,0 +1,15 @@
+            <field name="interval_number">12</field>
+            <field name="numbercall">-1</field>
+            <field name="model_id" ref="model_hr_employee"/>
+            <field name="function">_cron_compute_annual_balance</field>
```
Verify if it work:
2020-01-24 10:33:17,166 3357 WARNING v11c_hr_attendance odoo.models: ir.cron.create() includes unknown fields: function
Seem to be replace by the field "code":
https://webkul.com/blog/creating-cron-server-action-odoo-11/

eicher31 commented on hr_attendance_management/data/attendance_computation_cron.xml:
```diff
@@ -0,0 +1,12 @@
+            <field name="interval_type">days</field>
+            <field name="numbercall">-1</field>
+            <field name="model_id" ref="model_hr_employee"/>
+            <field name="function">_cron_create_attendance</field>
```
Same

eicher31 commented on hr_attendance_management/views/hr_employee.xml:
```diff
@@ -0,0 +1,87 @@
+               ref="hr.hr_kanban_view_employees"/>
+        <field name="arch"
+               type="xml">
+            <xpath expr="//div[@class='oe_kanban_details']/ul/li" position="after">
```
Use the hasclass(*classes) function to filter elements by their classes

eicher31 reviewed (changes requested):
FAIL: test_period_balances (odoo.addons.hr_attendance_management.tests.test_periods.TestPeriod)                                                   
Traceback (most recent call last):                                                                                                                
`   File "/home/stephane/odoo/odoo11/review/compassion-modules/hr_attendance_management/tests/test_periods.py", line 224, in test_period_balances 
`     self.assertEquals(self.gilles.balance, 4)                                                                                                   
` AssertionError: 5.0 != 4

ecino commented:
I think you should rebase onto branch 11.0 in order to fix the tests.

eicher31 commented on hr_attendance_management/tests/test_periods.py:
```diff
@@ -0,0 +1,427 @@
+            'lost': 0,
+            'employee_id': cls.jack.id,
+            'continuous_cap': True,
+            'origin': "create"
```
WARNING v11c_hr_attendance_test odoo.models: hr.employee.period.create() includes unknown fields: origin

eicher31 commented on hr_attendance_management/tests/test_periods.py:
```diff
@@ -0,0 +1,427 @@
+            'lost': 0,
+            'employee_id': cls.jack.id,
+            'continuous_cap': True,
+            'origin': "create"
```
WARNING v11c_hr_attendance_test odoo.models: hr.employee.period.create() includes unknown fields: origin

eicher31 commented on hr_attendance_management/tests/test_periods.py:
```diff
@@ -0,0 +1,427 @@
+            'lost': 0,
+            'employee_id': cls.jack.id,
+            'continuous_cap': True,
+            'origin': "create"
```
same

eicher31 reviewed (changes requested):
Move the hr_attendance settings inside the general settings and update the view. Some fields have not label.

quentingigon committed:
applied change requests + changed hr settings location. still a bug with the interface

quentingigon commented:
There is a problem with advanced_translation: it seems it can't launch the tests and after that the execution stops. Is it a problem with Travis ?

quentingigon committed:
removed a dependency on partner-contact-gender that was causing Travis to fail + tests corrections

quentingigon commented:
I removed a dependency on partner-contact-gender as it was causing Travis to fail. This module did add the gender field to res.partner but the values were (male, Male) instead of (m, Male). I replaced the dependency by thankyou_letter, as it's this module that add the gender field
